### ISSUE TITLE
Error parsing >4.2GB file on Windows

### ISSUE THREAD

**Issue by vinhdizzo** [WizardMac/ReadStat#36]
I'm getting the follow error for a file with 4855062528 bytes (errors out at about 30sec-1min). I am using RRO 3.2.2 (64 bit) with the latest haven and ReadStat from [this](https://github.com/WizardMac/ReadStat/issues/34) issue.
```{r}
> system.time(d1 <- read_sas('bar.sas7bdat')) / 60
Error: Failed to parse bar.sas7bdat: Invalid file, or file has unsupported features.
Timing stopped at: 0.07 0.28 35.03 
```
Data set could be generated from
```{sas}
data bar ;
	call streaminit(123);
	array x{10000} x1-x10000 ;
	do i=1 to 60000 ;
		do j=1 to dim(x) ;
			x(j) = rand('Uniform') ;
		end ;
		output ;
	end ;
run ;
```

**Comment by vinhdizzo** [comment]
Toyed with the sample size, and if I bring it down to 54000 rows (4369686528 bytes), the issue replicates; it does not at 53000 rows (4288790528 bytes).

**Comment by evanmiller** [comment]
Windows I assume?

**Comment by evanmiller** [comment]
Try this: [readstat_sas.c](https://raw.githubusercontent.com/WizardMac/ReadStat/7aa20258d4879f0230ac175fbdfcc52485c54163/src/readstat_sas.c)

**Comment by vinhdizzo** [comment]
Yes, I meant to put Windows in the original post.

Just tried and I get the same error.

**Comment by evanmiller** [comment]
I'm guessing this is another 32-bit/64-bit issue of some kind. (2^32 bytes = 4.2 GB). I'll post back when I have changes for you to test.

**Comment by vinhdizzo** [comment]
Yea, I had a similar hunch when I was trying to narrow down the file size, and it just happens to be 4.2GB.  Let me know!

**Comment by evanmiller** [comment]
This will produce a few debug messages: [readstat_sas.c](https://raw.githubusercontent.com/WizardMac/ReadStat/9687d2c61731fd54df030a3fb9216a204a4490d9/src/readstat_sas.c)

**Comment by vinhdizzo** [comment]
```{r}
> d1 = read_sas('bar.sas7bdat')
Reading header...

Looking for META and MIX pages...

Looking for AMD pages...

Error parsing page 53097 (Pass 1)

Error: Failed to parse bar.sas7bdat: Invalid file, or file has unsupported features.
```

**Comment by evanmiller** [comment]
Try these:

[readstat_sas.c](https://raw.githubusercontent.com/WizardMac/ReadStat/1f17a92d04e0f824886cb42558d09aaf987825b7/src/readstat_sas.c)
[readstat_io.c](https://raw.githubusercontent.com/WizardMac/ReadStat/2554d9d0d8883364df748dda7e8cbec3c4aec64c/src/readstat_io.c)

**Comment by vinhdizzo** [comment]
```{r}
> d1 = read_sas('bar.sas7bdat')
Error parsing page 53097 (Pass 1, pos = 456704)

Error: Failed to parse bar.sas7bdat: Invalid file, or file has unsupported features.
```

**Comment by evanmiller** [comment]
Ok, definitely looks like there's a 32-bit wraparound somewhere.

Try this: [readstat_io.h](https://raw.githubusercontent.com/WizardMac/ReadStat/309368435839be91ebbf874ea00c2d5acb6b9c9e/src/readstat_io.h)

**Comment by vinhdizzo** [comment]
Bingo.  Fixed it for the 4.2GB.  Tried a 9GB file (9708822528 bytes) as well.  All good.

**Comment by evanmiller** [comment]
Great to hear. Does this fix #35 as well?
